New obj and subj

howareyouman opened the pull request:
New parser for objects and subjects. Now we get not only single words, but also multi-word entities.

howareyouman committed:
modified:   src/main/java/com/w6/nlp/GetDoerAndVictim.java

howareyouman committed:
new file:   src/main/java/com/w6/data/Node.java
	modified:   src/main/java/com/w6/nlp/GetDoerAndVictim.java

towelenee commented on src/main/java/com/w6/data/Node.java:
```diff
@@ -0,0 +1,25 @@
+    String name;
+    List<Pair<String,String>> edges;
+    
+    public Node(String nameString, List<Pair<String,String>> ed)
```
Why here is not a Node? It will make object self contained.

towelenee commented on src/main/java/com/w6/nlp/GetDoerAndVictim.java:
```diff
@@ -28,23 +25,118 @@ private static void getResultWithViolentVerbs(
+            String word1, word2, tag, wordAndPos1,wordAndPos2;
+            Integer pos1,pos2;
+            
+            word1 = obj.gov().value();
```
I want better naming. word1 and word2 does give me information

towelenee commented on src/main/java/com/w6/nlp/GetDoerAndVictim.java:
```diff
@@ -28,23 +25,118 @@ private static void getResultWithViolentVerbs(
+            
+            word2 = obj.dep().value();
+            pos2 = obj.dep().index();
+            wordAndPos2 = word2+'\t' + pos2;
```
Why don't you use pairs?

towelenee commented on src/main/java/com/w6/nlp/GetDoerAndVictim.java:
```diff
@@ -28,23 +25,118 @@ private static void getResultWithViolentVerbs(
+        return result;
+    }
+    
+    static class MyStringComparator implements Comparator<String>{
```
MyStringComparator is a ad name. Why is static?

howareyouman replied:
Because it is used in static class.

towelenee commented on src/main/java/com/w6/nlp/GetDoerAndVictim.java:
```diff
@@ -28,23 +25,118 @@ private static void getResultWithViolentVerbs(
+        
+        for(Pair<String,String> node : nodeOfWord.getAllEdges())
+        {
+            List<String> tmp = getAllChilds(node.second, mapOfNodes);
```
tmp is a bad name

towelenee commented on src/main/java/com/w6/nlp/GetDoerAndVictim.java:
```diff
@@ -28,23 +25,118 @@ private static void getResultWithViolentVerbs(
+        for (String string : result.objects)
+        {
+            List<String> childs = getAllChilds(string,mapOfNodes);
+            Collections.sort(childs, new MyStringComparator());
```
Use pairs and you will not need comparator

howareyouman committed:
Added tests

Tests of object and subject were corrected for new parser and commented

towelenee commented on src/main/java/com/w6/nlp/Parser.java:
```diff
@@ -59,25 +69,41 @@ public Response generateResponse(final String input) {
-            sentenseWhat = violentVerbsParser.getAllViolentVerbs(parse);
-        
+            
+            int weightOfSentance = 1;
```
sentense

towelenee commented on src/test/java/ObjectAndSubjectTest.java:
```diff
@@ -2,57 +2,92 @@
-    /*
-
+    
+/*
```
Test is commented out

howareyouman committed:
Obj and Subj correction

Added new names of params, added pairs instead of string parse. Tests
checked.

towelenee commented on src/main/java/com/w6/nlp/GetDoerAndVictim.java:
```diff
@@ -28,23 +25,119 @@ private static void getResultWithViolentVerbs(
+            String wordGov, wordDep, tag;
+            Integer posGov, posDep;
+            
+            wordGov = obj.gov().value();
```
Still bad name

towelenee commented on src/main/java/com/w6/nlp/GetDoerAndVictim.java:
```diff
@@ -28,23 +25,119 @@ private static void getResultWithViolentVerbs(
+            tag = obj.reln().toString();
+            
+            
+            if(!mapOfNodes.containsKey(govStruct))
```
Copypasting code is usually bad ide

towelenee commented on src/main/java/com/w6/nlp/GetDoerAndVictim.java:
```diff
@@ -28,23 +25,119 @@ private static void getResultWithViolentVerbs(
+        
+    }
+    
+    private static String getNewWord(List<Pair<String, Integer>> words)
```
What does newWord mean?

towelenee commented on src/main/java/com/w6/nlp/GetDoerAndVictim.java:
```diff
@@ -28,23 +25,119 @@ private static void getResultWithViolentVerbs(
+        {
+            List<Pair<String, Integer>> childs = getAllChilds(word,mapOfNodes);
+            Collections.sort(childs, new ComparatorOfWords());
+            String newObject = getNewWord(childs);
```
Stop copypasting

towelenee commented on src/main/java/com/w6/nlp/GetDoerAndVictim.java:
```diff
@@ -28,23 +25,119 @@ private static void getResultWithViolentVerbs(
+        
+        HashMap<Pair<String, Integer>, Node> mapOfNodes = new HashMap<>();
+        
+        for (TypedDependency obj:list)
```
obje and list are bad names

towelenee commented on src/main/java/com/w6/nlp/GetDoerAndVictim.java:
```diff
@@ -28,23 +25,119 @@ private static void getResultWithViolentVerbs(
+        ArrayList<String> newObjectList = new ArrayList<>();
+        ArrayList<String> newSubjectList = new ArrayList<>();
+        
+        for (Pair<String, Integer> word : listOfSubjects)
```
should we make it a function?

howareyouman committed:
Total refactor

Added 2 new methods to remove copy-past, renamed 1 method.

towelenee commented on src/main/java/com/w6/nlp/GetDoerAndVictim.java:
```diff
@@ -28,39 +25,111 @@ private static void getResultWithViolentVerbs(
-                
+            String tag = dependency.reln().toString();
+            
+            Pair<String, Integer> govStruct = new Pair(dependency.gov().value(), dependency.gov().index());
```
not gov but governer or better main word.

howareyouman committed:
Names correction

howareyouman committed:
Preparing for mock testing

towelenee commented on src/main/java/com/w6/nlp/Parser.java:
```diff
@@ -110,17 +136,93 @@ public Response generateResponse(final String input) {
+    
+    class WordComparator implements Comparator<Pair<String, Double>>{
+        @Override
+        public int compare(Pair<String, Double> a, Pair<String, Double> b) {
```
lhs and rhs

howareyouman committed:
rename

howareyouman committed:
deleted:    src/main/java/com/w6/data/ObjectsAndSubjects.java
	new file:   src/main/java/com/w6/nlp/DependencyTree.java
	modified:   src/main/java/com/w6/nlp/GetDoerAndVictim.java
	modified:   src/main/java/com/w6/nlp/Parser.java
	modified:   src/test/java/ObjectAndSubjectTest.java

howareyouman committed:
modified:   src/main/java/com/w6/nlp/GetDoerAndVictim.java
	modified:   src/main/java/com/w6/nlp/LocationParser.java
	modified:   src/main/java/com/w6/nlp/Parser.java

howareyouman committed:
refactor

howareyouman committed:
refactor

howareyouman committed:
Merge branch 'master' of github.com:W6-Assess/dev into newObjAndSubj

howareyouman committed:
Merge branch 'newObjAndSubj' of github.com:W6-Assess/dev into newObjAndSubj

howareyouman committed:
new file:   src/main/java/com/w6/data/CollectionOfWords.java
	modified:   src/main/java/com/w6/data/Word.java
	modified:   src/main/java/com/w6/nlp/DependencyTree.java

howareyouman committed:
Refactor with collections

Added collections DoerAndSubject and Location.
Small refactor of DependencyTree and Word.
